sklearn/ensemble/tests/test_gradient_boosting.py

```diff
@@ -544,71 +550,133 @@ def test_shape_y():
     y_ = np.asfortranarray(y_)
-    clf = GradientBoostingClassifier(n_estimators=100, random_state=1)
-    clf.fit(X, y_)
-    assert_array_equal(clf.predict(T), true_result)
-    assert 100 == len(clf.estimators_)
+    estimator = GradientBoostingClassifier(n_estimators=100, random_state=1)
+    estimator.fit(X, y_)
+    assert_array_equal(estimator.predict(T), true_result)
+    assert 100 == len(estimator.estimators_)
 
 
-def test_oob_improvement():
+@pytest.mark.parametrize("GradientBoostingEstimator", GRADIENT_BOOSTING_ESTIMATORS)
+def test_oob_improvement(GradientBoostingEstimator):
     # Test if oob improvement has correct shape and regression test.
-    clf = GradientBoostingClassifier(n_estimators=100, random_state=1, subsample=0.5)
-    clf.fit(X, y)
-    assert clf.oob_improvement_.shape[0] == 100
+    estimator = GradientBoostingEstimator(
+        n_estimators=100, random_state=1, subsample=0.5
+    )
+    estimator.fit(X, y)
+    assert estimator.oob_improvement_.shape[0] == 100
     # hard-coded regression test - change if modification in OOB computation
     assert_array_almost_equal(
-        clf.oob_improvement_[:5], np.array([0.19, 0.15, 0.12, -0.12, -0.11]), decimal=2
+        estimator.oob_improvement_[:5],
+        np.array([0.19, 0.15, 0.12, -0.11, 0.11]),
+        decimal=2,
+    )
+
+
+@pytest.mark.parametrize("GradientBoostingEstimator", GRADIENT_BOOSTING_ESTIMATORS)
+def test_oob_scores(GradientBoostingEstimator):
+    # Test if oob scores has correct shape and regression test.
+    X, y = datasets.make_hastie_10_2(n_samples=100, random_state=1)
+    estimator = GradientBoostingEstimator(
+        n_estimators=100, random_state=1, subsample=0.5
     )
+    estimator.fit(X, y)
+    assert estimator.oob_scores_.shape[0] == 100
+    assert estimator.oob_scores_[-1] == estimator.oob_score_
+
+    estimator = GradientBoostingEstimator(
+        n_estimators=100,
+        random_state=1,
+        subsample=0.5,
+        n_iter_no_change=5,
+    )
+    estimator.fit(X, y)
+    assert estimator.oob_scores_.shape[0] < 100
+    assert estimator.oob_scores_[-1] == estimator.oob_score_
+
+
+@pytest.mark.parametrize("GradientBoostingEstimator", GRADIENT_BOOSTING_ESTIMATORS)
+def test_no_available_oob_with_full_sample(GradientBoostingEstimator):
+    """Check that not OOB scores attribute are available with `subsamples=1.0`."""
+    X, y = datasets.make_hastie_10_2(n_samples=100, random_state=1)
+    estimator = GradientBoostingEstimator(
+        n_estimators=100,
+        random_state=1,
+        subsample=1.0,
+    )
+    estimator.fit(X, y)
+    assert not hasattr(estimator, "oob_scores_")
+    assert not hasattr(estimator, "oob_score_")
 
 
-def test_oob_improvement_raise():
-    # Test if oob improvement has correct shape.
-    clf = GradientBoostingClassifier(n_estimators=100, random_state=1, subsample=1.0)
-    clf.fit(X, y)
+@pytest.mark.parametrize("GradientBoostingEstimator", GRADIENT_BOOSTING_ESTIMATORS)
```

Comment:
Now I am seeing this test. Could you parametrize it to check the attributes `oob_scores_` and `oob_score_` as well instead of writing a new test. Then, let's keep the `pytest.raises`.